docs/src/implementation.md

```diff
@@ -1,16 +1,21 @@
 # Implementation details
 
-`CategoricalArray` is based on the `CategoricalPool` type, which keeps track of the levels and associates them with an integer reference (for internal use). It offers methods to set levels, change their order while preserving the references, and efficiently get the integer index corresponding to a level and vice-versa. It is also parameterized on the type used to store the references, so that small pools can use as little memory as possible. Finally, it keeps a vector of value objects (`CategoricalValue`), so that `getindex` can return the existing object instead of allocating a new one.
+`CategoricalArray` is made of the two fields:
 
-`CategoricalArray` is made of two fields:
+- `refs`: an integer array that stores the indices of levels for each `CategoricalArray` element; `0` denotes a missing value (for `CategoricalArray{Union{T, Null}}` only).
```

Comment:
"indices of levels" could be misleading since these are not the positions of the elements in the levels, but in the index. I"d rather say something like "that stores the position of the level in the pool's index for each element". Maybe we should also point to the text below which describes the index.

Reply:
btw, the corresponding field and method in `CategoricalString/CategoricalValue` are called `level`. To avoid confusion I would rather call them either `ref` (it might be confused with `Base.Ref`, but it's consistent with `reftype()` and `CategoricalArray.refs`) or `index`.